src/mlpack/tests/main_tests/logistic_regression_test.cpp

```diff
@@ -0,0 +1,378 @@
+  SetInputParam("test", std::move(testX));
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+
+}
+
+/**
+ * Ensuring that test data dimensionality is checked when model is loaded.
+ */
+BOOST_AUTO_TEST_CASE(LRWrongDimOfTestData2)
+{
+  constexpr int N = 10;
+  constexpr int D = 3;
+  constexpr int M = 15;
+
+  arma::mat trainX = arma::randu<arma::mat>(D, N);
+  arma::Row<size_t> trainY;
+
+  trainY << 0 << 1 << 0 << 1 << 1 << 1 << 0 << 1 << 0 << 0 << arma::endr; // 10 responses
+
+  SetInputParam("training", std::move(trainX));
+  SetInputParam("labels", std::move(trainY));
+
+  // training the model
+  mlpackMain();
+
+  LogisticRegression<> model = CLI::GetParam<LogisticRegression<>>("output_model");
+
+  //reset the settings
+  CLI::ClearSettings();
+  CLI::RestoreSettings(testName);
+
+  arma::mat testX = arma::randu<arma::mat>(D - 1, M); // test data with Wrong dimensionality.
+  SetInputParam("input_model", std::move(model));
+  SetInputParam("test", std::move(testX));
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+/**
+  * Ensuring that training responses contain only two classes (0 or 1)
+ **/
+BOOST_AUTO_TEST_CASE(LRTrainWithMoreThanTwoClasses) 
+{
+  constexpr int N = 8;
+  constexpr int D = 2;
+
+  arma::mat trainX = arma::randu<arma::mat>(D,N);
+  arma::Row<size_t> trainY;
+
+  trainY << 0 << 1 << 0 << 1 << 2 << 1 << 3 << 1 << arma::endr; // 8 responses containing more than two classes
+
+  SetInputParam("training", std::move(trainX));
+  SetInputParam("labels", std::move(trainY));
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(),std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+/**
+  * Ensuring that max iteration for optimizers is non negative 
+ **/ 
+BOOST_AUTO_TEST_CASE(LRNonNegativeMaxIterationTest) 
+{
+  constexpr int N = 10;
+  constexpr int D = 3;
+
+  arma::mat trainX = arma::randu<arma::mat>(D,N);
+  arma::Row<size_t> trainY;
+
+  trainY << 0 << 1 << 0 << 1 << 1 << 1 << 0 << 1 << 0 << 0 << arma::endr; // 10 responses
+
+  SetInputParam("training", std::move(trainX));
+  SetInputParam("labels", std::move(trainY));
+  SetInputParam("max_iterations", int(-1));
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+
+}
+
+/**
+  * Ensuring that max_iterations for optimizers is integer value
+ **/
+BOOST_AUTO_TEST_CASE(LRIntegerMaxIterationTest)
+{
+  constexpr int N = 10;
+  constexpr int D = 3;
+
+  arma::mat trainX = arma::randu<arma::mat>(D,N);
+  arma::Row<size_t> trainY;
+
+  trainY << 0 << 1 << 0 << 1 << 0 << 1 << 0 << 1 << 0 << 1 << arma::endr; // 10 responses
+
+  SetInputParam("training", std::move(trainX));
+  SetInputParam("labels", std::move(trainY));
+  SetInputParam("max_iterations", int(0.01));
```

Comment:
I'm a bit confused about this one, shouldn't `int(0.01)` be 0?  0 is an allowable value for `max_iterations`, it means "no limit".

Reply:
Shouldn't int(0.01) value of max iterations give an error as it is an non integer value instead converting it into int type i.e, 0.

Reply:
No, when you write `int(x)` that is an explicit conversion to type `int`, so in your case `int(0.01)` will create an `int` with value `0`.

Reply:
Yes. I did that mistake. I should use double instead of int.

Reply:
I wouldn't suggest that, because that will cause `SetInputParam` to be called with a template type of `double`, which will cause all kinds of other disasters.  You can assume that whenever `mlpackMain()` is called, every parameter is of the right type.  Even if I do the following:

```
CLI::GetParam<int>("max_iterations") = double(0.01);
```

if it compiles (and it will at least issue a warning), the double will be implicitly cast to an int, resulting in the `0` value.

Reply:
Ok. So I don't think this test case is needed.

Reply:
Right, I would agree with that.